pandas/core/reshape/merge.py

```diff
@@ -1526,6 +1528,11 @@ def _maybe_coerce_merge_keys(self) -> None:
             ) or (lk.dtype.kind == "M" and rk.dtype.kind == "M"):
                 # allows datetime with different resolutions
                 continue
+            # datetime and timedelta not allowed
+            elif is_datetime64_dtype(lk.dtype) and is_timedelta64_dtype(rk.dtype):
```

Comment:
Can you check the `dtype.kind` instead of using these methods?

Reply:
Sure! I'll check `dtype.kind` to keep consistent with the logic above.